sdcm/nemesis.py

```diff
@@ -431,29 +437,35 @@ def approach_enospc():
                 self.log.error(str(details))
             return search_database_enospc(node) > orig_errors
 
-        for node in nodes:
-            result = node.remoter.run('cat /proc/mounts')
-            if '/var/lib/scylla' not in result.stdout:
-                self.log.error("Scylla doesn't use an individual storage, skip enospc test")
-                continue
-
-            # check original ENOSPC error
-            orig_errors = search_database_enospc(node)
-            wait.wait_for(func=approach_enospc,
-                          timeout=300,
-                          step=5,
-                          text='Wait for new ENOSPC error occurs in database')
-
-            self.log.debug('Sleep {} seconds before releasing space to scylla'.format(sleep_time))
-            time.sleep(sleep_time)
-
-            self.log.debug('Delete occupy_90percent file to release space to scylla-server')
-            node.remoter.run('sudo rm -rf /var/lib/scylla/occupy_90percent.*')
-
-            self.log.debug('Sleep a while before restart scylla-server')
-            time.sleep(sleep_time / 2)
-            node.remoter.run('sudo systemctl restart scylla-server.service')
-            node.wait_db_up()
+        enospc_log_filter = EventsFilter(name='filter_enospc', filter_name='NO_SPACE_ERROR')
+        enospc_backtrace_filter = EventsFilter(name='filter_enospc_backtrace', filter_name='BACKTRACE', filter_string='No space left on device')
```

Comment:
API: EventsFilter(type=..., string=)
remove `name` argument and use `id` of the object to identify in list of events

Reply:
ADD `contextmanager` support